cylc/uiserver/handlers.py

```diff
@@ -145,9 +150,58 @@ def prepare(self):
         super().prepare()
 
 
+class SubscriptionHandler(websocket.WebSocketHandler):
+    subscription_server = None
+    resolvers = None
+    queue = Queue(100)
+
+    def initialize(self, sub_server, resolvers):
+        self.subscription_server = sub_server
+        self.resolvers = resolvers
+
+    def select_subprotocol(self, subprotocols):
+        return GRAPHQL_WS
+
+    def open(self, *args, **kwargs):
+        IOLoop.current().spawn_callback(self.subscription_server.handle, self,
+                                        self.context)
+
+    async def on_message(self, message):
+        await self.queue.put(message)
+
+    async def recv(self):
+        return await self.queue.get()
+
+    def check_origin(self, origin: str) -> bool:
+        return True
+
+    @property
+    def context(self):
+        wider_context = {
+            'request': self.request,
+            'resolvers': self.resolvers,
+        }
+        return wider_context
+
+
+class GraphiQLHandler(UIServerGraphQLHandler):
```

Comment:
From the PR's example. This handler renders a custom React app that talks WebSockets. So that the query is reloaded automatically :+1: